src/utils/helper-functions.ts

```diff
@@ -0,0 +1,41 @@
+import { DeleteResult, UpdateResult } from 'typeorm';
+import {
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+
```

Comment:
read about response envelopes and how to use them in NestJS

Reply:
I searched for this topic but cannot reach resources. Could you please suggest a resource to read from?

Reply:
```
interface BaseResponse<T = Record<string, any>> {
  data: T | null;
  statusCode: number;
  headers?: Record<string, any>;
  path: string;
  method: string;
  timeStamp: Date;
}

export interface SuccessfulResponse extends BaseResponse {
  success: true;
  successMessage: string;
  successMessageKey: string;
}

export interface ErrorResponse extends BaseResponse {
  success: false;
  errorMessage: string;
  errorMessageKey: string;
}

export type Response = SuccessfulResponse | ErrorResponse;

```

use that file as a reference and when you want to create succesful response for an endpoint you would have to make it implement  SuccessfulResponse and that would help have a well defined response for all endpoints and hence easier integration for front-end

Reply:
Adding it